aspnetcore/grpc/basics.md

```diff
@@ -9,53 +9,60 @@ uid: grpc/basics
+[!code-xml[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/GrpcGreeter.Server.csproj?highlight=2&range=7-10)]
 
-## C# Tooling support for `.proto` files
+## C# Tooling support for .proto files
 
-The tooling package [Grpc.Tools](https://www.nuget.org/packages/Grpc.Tools/) is required to generate the C# assets from `.proto` files. The assets are generated on a as-needed basis every time the project is built. This dependency is required by both the server and client projects and can be added by using the Package Manager in Visual Studio or adding a `<PackageReference>` to your project file:
+The tooling package [Grpc.Tools](https://www.nuget.org/packages/Grpc.Tools/) is required to generate the C# assets from *.proto* files. The assets are generated on a as-needed basis each time the project is built. This dependency is required by both the server and client projects. `Grpc.Tools` can be added by using the Package Manager in Visual Studio or adding a `<PackageReference>` to the project file:
 
-[!code-xml[](~//tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/GrpcGreeter.Server.csproj?highlight=16)]
+[!code-xml[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/GrpcGreeter.Server.csproj?highlight=1&range=16)]
 
-Since the tooling package is not required at runtime, the dependency should be marked with `PrivateAssets="All"`.
+The tooling package is not required at runtime, therefore, the dependency should be marked with `PrivateAssets="All"`.
 
 ## Generated C# assets
 
-The tooling package will generate the C# types representing the messages defined in the included `.proto` files.
-
-For server side assets, an abstract service base type is generated. The base type contains the definitions of all the gRPC calls contained in the `.proto` file. A concrete service implementation derives from this base type and implements the logic for the gRPC calls. For the `greet.proto` example described above, an abstract `GreeterBase` type that contains a virtual `SayHello` method is generated. A concrete implementation `GreeterService` then overrides the method and implements the logic handling the gRPC call.
+The tooling package will generate the C# types representing the messages defined in the included *.proto* files.
 
-[!code-cs[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/Services/GreeterService.cs?highlight=10,12-18)]
+For server side assets, an abstract service base type is generated. The base type contains the definitions of all the gRPC calls contained in the *.proto* file. A concrete service implementation derives from this base type and implements the logic for the gRPC calls. For the `greet.proto` example described previously, an abstract `GreeterBase` type that contains a virtual `SayHello` method is generated. A concrete implementation `GreeterService` overrides the method and implements the logic handling the gRPC call.
```

Comment:
```suggestion
For server side assets, an abstract service base type is generated. The base type contains the definitions of all the gRPC calls contained in the *.proto* file. You then create a concrete service implementation derives from this base type and implements the logic for the gRPC calls. For the `greet.proto` example described previously, an abstract `GreeterBase` type that contains a virtual `SayHello` method is generated. A concrete implementation `GreeterService` overrides the method and implements the logic handling the gRPC call.
```

A lot of devs will get that you are reasonable for creating the implementation, but I think it is helpful to be explicit. Reword as needed